gsutil.py

```diff
@@ -73,24 +73,40 @@ def OutputAndExit(message):
     ('pyasn1-modules', ''),  # oauth2client dependency
     ('rsa', ''),  # oauth2client dependency
     ('apitools', ''),
-    ('boto', ''),
+    ('boto', ''), # XML API and AWS interop dependency
     ('gcs-oauth2-boto-plugin', ''),
     ('fasteners', ''), # oauth2client and apitools dependency
     ('monotonic', ''), # fasteners dependency
     ('httplib2', 'python2'),
     ('python-gflags', ''),
     ('retry-decorator', ''),
-    ('six', ''),
+    ('six', ''), # Python 2 / 3 compatibility dependency
     ('socksipy-branch', ''),
 ]
-for libdir, subdir in THIRD_PARTY_LIBS:
-  if not os.path.isdir(os.path.join(THIRD_PARTY_DIR, libdir)):
-    OutputAndExit(
-        'There is no %s library under the gsutil third-party directory (%s).\n'
-        'The gsutil command cannot work properly when installed this way.\n'
-        'Please re-install gsutil per the installation instructions.' % (
-            libdir, THIRD_PARTY_DIR))
-  sys.path.insert(0, os.path.join(THIRD_PARTY_DIR, libdir, subdir))
+
+# These are special third party libraries that live in
+# `gsutil/third_party/vendored`. These libraries have modifications specifically
+# for gsutil to operate smoothly that may not yet be present in the latest
+# release of the package maintainer's repositoy.
+VENDORED_THIRD_PARTY_LIBS = [
+    ('boto', 'vendored') # XML API and AWS interop dependency
```

Comment:
Reversed; boto is the subdir of vendored. 

Also, with the prepend_to_path() method, it could be changed so the dir being passed in is `gsutil/third_party/vendored/` so that 'vendored' doesn't have to be added to the tuple. This also comes in handy for adding future packages like crcmod where an additional subdir is needed to specify python2 or python3.

Reply:
Good suggestion, I'll pass the path including 'vendored' in to `prepend_to_path` so we'll only need to specify boto.